Kickstarter-iOS/Features/ReportProject/ReportProjectFormView.swift

```diff
@@ -8,45 +8,54 @@ enum ReportFormFocusField {
 
 @available(iOS 15.0, *)
 struct ReportProjectFormView: View {
-  let projectID: String
-  let projectURL: String
-  let projectFlaggingKind: GraphAPI.FlaggingKind
   @Binding var popToRoot: Bool
 
   @SwiftUI.Environment(\.dismiss) private var dismiss
-  @ObservedObject private var viewModel = ReportProjectFormViewModel()
+  @ObservedObject private var viewModel: ReportProjectFormViewModel
 
-  @State private var retrievedEmail = ""
-  @State private var details: String = ""
-  @State private var saveEnabled: Bool = false
-  @State private var saveTriggered: Bool = false
   @State private var showLoading: Bool = false
-  @State private var showBannerMessage = false
-  @State private var submitSuccess = false
-  @State private var bannerMessage: MessageBannerViewViewModel?
```

Comment:
Yeah, I think that's a really interesting question, and I'm not sure I have a great answer for it, either. The way we've structured things here, the viewmodel actually has _no knowledge_ of the view directly, so at  there would be no way for it to even ask the view for its `detailsText`. 

I think `@Binding` kind of raises some interesting questions about ownership in general, since it's really a two-way relationship between the model and the UI. It would be really interesting to try and find some other examples of MVVM in SwiftUI and see what other patterns folks have cooked up.

Reply:
At present, my understanding of `@State` is that it's for a variable that is owned by a view, and/or used by its subviews. At first glance to me, it seems as if we don't have much in the way of state that is ever truly owned by a view - unless it was something like, say, a tab collapsing/expanding.

Reply:
I agree with this - the default should be for the viewmodel to own things (unless it's literally something that only affects the UI). Any time logic can live in the viewmodel instead of the view, it should.